lms/djangoapps/grades/management/commands/tests/test_reset_grades.py

```diff
@@ -0,0 +1,289 @@
+
+    def _assert_course_delete_stat_logged(self, mock_log, num_rows):
+        self._assert_stat_logged(mock_log, num_rows, PersistentCourseGrade, 'Deleted', log_offset=4)
+
+    def _assert_subsection_delete_stat_logged(self, mock_log, num_rows):
+        self._assert_stat_logged(mock_log, num_rows, PersistentSubsectionGrade, 'Deleted', log_offset=2)
+
+    def _assert_course_query_stat_logged(self, mock_log, num_rows, num_courses=None):
+        if num_courses is None:
+            num_courses = self.num_courses
+        log_offset = num_courses + 1 + num_courses + 1
+        self._assert_stat_logged(mock_log, num_rows, PersistentCourseGrade, 'Would delete', log_offset)
+
+    def _assert_subsection_query_stat_logged(self, mock_log, num_rows, num_courses=None):
+        if num_courses is None:
+            num_courses = self.num_courses
+        log_offset = num_courses + 1
+        self._assert_stat_logged(mock_log, num_rows, PersistentSubsectionGrade, 'Would delete', log_offset)
+
+    def _date_from_now(self, days=None):
+        return datetime.now() + timedelta(days=days)
+
+    def _date_str_from_now(self, days=None):
+        future_date = self._date_from_now(days=days)
+        return future_date.strftime(reset_grades.DATE_FORMAT)
+
+    @patch('lms.djangoapps.grades.management.commands.reset_grades.log')
+    def test_reset_all_courses(self, mock_log):
+        self._update_or_create_grades()
+        self._assert_grades_exist_for_courses(self.course_keys)
+
+        self.command.handle(delete=True, all_courses=True)
+
+        self._assert_grades_absent_for_courses(self.course_keys)
+        self._assert_subsection_delete_stat_logged(
+            mock_log,
+            num_rows=self.num_users * self.num_courses * self.num_subsections,
+        )
+        self._assert_course_delete_stat_logged(
+            mock_log,
+            num_rows=self.num_users * self.num_courses,
+        )
+
+    @patch('lms.djangoapps.grades.management.commands.reset_grades.log')
+    @ddt.data(1, 2, 3)
+    def test_reset_some_courses(self, num_courses_to_reset, mock_log):
+        self._update_or_create_grades()
+        self._assert_grades_exist_for_courses(self.course_keys)
+
+        self.command.handle(
+            delete=True,
+            courses=[unicode(course_key) for course_key in self.course_keys[:num_courses_to_reset]]
+        )
+
+        self._assert_grades_absent_for_courses(self.course_keys[:num_courses_to_reset])
+        self._assert_grades_exist_for_courses(self.course_keys[num_courses_to_reset:])
+        self._assert_subsection_delete_stat_logged(
+            mock_log,
+            num_rows=self.num_users * num_courses_to_reset * self.num_subsections,
+        )
+        self._assert_course_delete_stat_logged(
+            mock_log,
+            num_rows=self.num_users * num_courses_to_reset,
+        )
+
+    def test_reset_by_modified_start_date(self):
```

Comment:
Good catch.  For certain issues, we will want to search just by `modified`, without first filtering by`course_id`.  Given that, I do agree we should consider adding an index on the `modified` column.  And yes, is it not better to do so sooner than later, before the table gets bigger?